cmd/fs-v1-multipart.go

```diff
@@ -504,6 +504,7 @@ func (fs *FSObjects) CompleteMultipartUpload(ctx context.Context, bucket string,
 	if _, err := fs.statBucketDir(ctx, bucket); err != nil {
 		return oi, toObjectErr(err, bucket)
 	}
+	defer ObjectPathUpdated(pathutil.Join(bucket, object))
```

Comment:
We have pathJoin() also, which is perfect because it doesn't remove the trailing slash in object (empty directory object)

Reply:
Not sure if I will call it "perfect". It is not needed here and stdlib > homewritten stuff.

Reply:
> Not sure if I will call it "perfect". It is not needed here and stdlib > homewritten stuff.

One problem with path.Join() is that it removes the trailing slash of any path
For example, joining "bucket" and "object/" will result "bucket/object", but this is wrong, the object name should be "object/" and not "object" (this is a special and possible case in S3 spec)

Reply:
In this case it doesn't matter since you cannot have an object and a folder with the same name. I want to be harder to send misleading data, so even if somebody accidentally strips the `/` it will still work.

The trailing slash is trimmed consistently for everything in the crawler, so I definitely don't want to keep it.